ecommerce/taxes/lib/taxes/vat_rate_catalog.rb

```diff
@@ -16,13 +16,17 @@ def vat_rate_for(product_id)
     end
 
     def vat_rate_by_code(vat_rate_code)
-      @event_store
+      last_available_vat_rate_event = @event_store
         .read
         .stream("Taxes::AvailableVatRate$#{vat_rate_code}")
         .last
-        &.data
-        &.fetch(:vat_rate)
-        &.then { |vat_rate| Infra::Types::VatRate.new(vat_rate) }
+
+      if last_available_vat_rate_event.instance_of?(AvailableVatRateAdded)
```

Comment:
If you only care about the `AvailableVatRateAdded` event then you could use RES client's api to get that:
`@event_store.read.stream("Taxes::AvailableVatRate$#{vat_rate_code}").of_type(AvailableVatRateAdded).last`

the if statement wouldn't be necessary

Reply:
I actually care that the last event is `AvailableVatRateAdded` and not `AvailableVatRateRemoved` to ensure that this vat rate wasn’t removed.